libraries-3/src/main/java/com/baeldung/mail/attachment/EmailAttachmentReceiver.java

```diff
@@ -0,0 +1,108 @@
+import javax.mail.Store;
+import javax.mail.internet.MimeBodyPart;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EmailAttachmentReceiver {
+
+    final Logger LOGGER = LoggerFactory.getLogger(EmailAttachmentReceiver.class);
+
+    private static final String MAIL_PROTOCOL_PORT = "mail.%s.port";
+    private static final String MAIL_PROTOCOL_HOST = "mail.%s.host";
+    private static final String MAIL_PROTOCOL_SOCKET_FACTORY_PORT = "mail.%s.socketFactory.port";
+    private static final String MAIL_PROTOCOL_SOCKET_FACTORY_FALLBACK = "mail.%s.socketFactory.fallback";
+    private static final String MAIL_PROTOCOL_SOCKET_FACTORY_CLASS = "mail.%s.socketFactory.class";
+    private static final String FALSE = "false";
+    private static final String JAVAX_NET_SSL_SSL_SOCKET_FACTORY = "javax.net.ssl.SSLSocketFactory";
+    private static final String MULTIPART = "multipart";
+    private static final String INBOX = "INBOX";
+
+    private String protocol;
+    private String host;
+    private String port;
+    private String userName;
+    private String password;
+
+    public EmailAttachmentReceiver(String host, String protocol, String port, String userName, String password) {
+        this.protocol = protocol;
+        this.host = host;
+        this.port = port;
+        this.userName = userName;
+        this.password = password;
+    }
+
+    public void downloadEmailAttachmentsTo(String directory) {
+        Properties properties = new Properties();
+
+        // server settings
+        properties.put(String.format(MAIL_PROTOCOL_HOST, this.protocol), this.host);
+        properties.put(String.format(MAIL_PROTOCOL_PORT, this.protocol), this.port);
+
+        // SSL settings
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_CLASS, this.protocol), JAVAX_NET_SSL_SSL_SOCKET_FACTORY);
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_FALLBACK, this.protocol), FALSE);
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_PORT, this.protocol), this.port);
+
+        // obtaining sessions
+        Session session = Session.getDefaultInstance(properties);
+        try {
+                // connecting to the mail server
+                Store store = session.getStore(this.protocol);
+                store.connect(this.userName, this.password);
+
+            // opens the inbox folder
+            Folder inbox = store.getFolder(INBOX);
+            inbox.open(Folder.READ_ONLY);
+            List<Message> messages = asList(inbox.getMessages());
+            for (Message message : messages) {
+                // filtering message with content type 'multipart'
+                if (message.getContentType()
+                    .equalsIgnoreCase(MULTIPART)) {
+                    checkForAttachmentsAndSave(message, directory);
+                }
+            }
+            // closing open connections
+            inbox.close();
+            store.close();
```

Comment:
We're not using `try-with-resources`. We could do. If not, then these `close` methods would have to be in the `finally` block in case of exceptions. The mail API can throw a lot of exceptions.

I'd recommend using the try-with resources. `Folder` is definitely `AutoCloseable`